src/Arrays/Programs_involving_Data_Sequences/Hard/Q1/CountArTest.java

```diff
@@ -0,0 +1,73 @@
+package Arrays.Programs_involving_Data_Sequences.Hard.Q1;
+
+import global.BaseTest;
+import global.variables.*;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CountArTest extends BaseTest{
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new IntegerLiteral(0),
+                new NewLine(),
+                new IntegerLiteral(1),
+                new NewLine(),
+                new IntegerLiteral(0),
+                new NewLine(),
+                new IntegerLiteral(2),
+                new NewLine(),
+                new IntegerLiteral(3),
+                new NewLine(),
+                new IntegerLiteral(1),
+                new NewLine(),
+                new IntegerLiteral(0),
+                new NewLine(),
+                new IntegerLiteral(2),
+                new NewLine(),
+                new IntegerLiteral(0),
+                new NewLine(),
+                new IntegerLiteral(3),
+        };
+    }
+
+    public void runMain() {
+        CountAr.main(new String[]{});
+    }
+
+    @RepeatedTest(50)
+    public void methodTestRandomNums(){
+        int[] expected = new int[10];
+        int[] passed = new int[10];
+        for(int x = 0; x <10; x++){
+            int n = (int)(Math.random()*10);
+            passed[x] = n;
+            expected[n]++;
+        }
+        int[] actual = CountAr.countThisPlease(passed);
+
+        assertArrayEquals(expected, actual, "Expected output: " + Arrays.toString(expected) + " Received output: " + Arrays.toString(actual));
```

Comment:
This error message would probably be confusing for a student to interpret what was actually wrong. Perhaps "Your method did not correctly count for the array [the array]".